### PR TITLE
Simplifying our dev dockerfile

### PR DESCRIPTION
**Pull request by parallelo**

### CODE REVIEW

File: tensorflow/tools/docker/Dockerfile.devel-rocm [hunk 1]
```diff
@@ -205,7 +135,7 @@ RUN mkdir /bazel && \
 
 # Download and build TensorFlow.
 WORKDIR /tensorflow-upstream
-RUN git clone --branch=.* --depth=1 https://github.com/ROCmSoftwarePlatform/tensorflow-upstream.git .
+RUN git clone --branch=r1.10-rocm --depth=1 https://github.com/ROCmSoftwarePlatform/tensorflow-upstream.git .
```
Comment (whchung) [inline tensorflow/tools/docker/Dockerfile.devel-rocm]: @parallelo any reason we stick with `r1.10-rocm` branch here, instead of `develop-upstream`?

Comment (parallelo) [reply tensorflow/tools/docker/Dockerfile.devel-rocm]: That's a good question.  The other Dockerfiles (e.g. `Dockerfile.devel-gpu`) are already pointing to r1.11, so I just used the most recent TF release for ROCm.  I'm open to changing this though.

Comment (whchung) [reply tensorflow/tools/docker/Dockerfile.devel-rocm]: could you help clarify the purpose of the dockerfile? if it's deemed to stick with a particular released version, it's fine to use `r1.10-rocm`. or if it's always the cutting edge tip we need, maybe we should use `develop-upstream`.

Comment (sunway513) [reply tensorflow/tools/docker/Dockerfile.devel-rocm]: To check against ROCm staging bits, we should use a TF release branch.

Comment (whchung) [reply tensorflow/tools/docker/Dockerfile.devel-rocm]: ok i see. then `r1.10-rocm` it is.

**Pull request merged**